src/main/haskell/kore/src/Kore/Step/Search.hs

```diff
@@ -128,32 +127,45 @@ matchWith
         , FreshVariable variable
         , Hashable variable
         , Ord (variable level)
-        , Ord (variable Meta)
-        , Ord (variable Object)
+        , OrdMetaOrObject variable
         , Show (variable level)
-        , Show (variable Meta)
-        , Show (variable Object)
+        , ShowMetaOrObject variable
         )
     => MetadataTools level StepperAttributes
     -> PredicateSubstitutionSimplifier level Simplifier
     -> PureMLPatternSimplifier level variable
     -> ExpandedPattern level variable
     -> ExpandedPattern level variable
-    -> MaybeT Simplifier (UnificationSubstitution level variable)
+    -> MaybeT Simplifier (PredicateSubstitution level variable)
 matchWith tools substitutionSimplifier simplifier e1 e2 = do
     (unifier, _proof) <- hushT (unificationProcedure tools substitutionSimplifier t1 t2)
-    let (predicate, _proof) = give (MetadataTools.symbolOrAliasSorts tools)
-            $ makeMultipleAndPredicate
+    (predicate, _proof) <-
```

Comment:
I think that `predicate` also contains a substitution, please rename if so.

Reply:
I think you are right, I have a note that I forgot about. I'll double check and rename. Thanks!